### PR TITLE
First submission of layouts and templates readme lesson

### PR DESCRIPTION
**Pull request by hendrikswan**
Please keep in mind it's my first lesson, so I'm still getting used to the process :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,26 +1,197 @@
...
+
+Across all these views you would want a consistent look, which is called a layout. This layout perhaps contains something like a logo, navigation links, a search bar, and a footer at the bottom that contains some info about the shop.
+
+Without layouts you would need to copy and paste your logo, navigation, search, and footer code into each action's template that you want in your application.
+
+And when you want to make a change to one of the common components, for example the navigation, you would have to go and make that change in all the template files that you created in your application. This will obviously slow you down a lot.
+
+## Layouts To The Rescue
+
+Luckily you don't need to copy content from one template file to the next, because layouts in Rails are enabled by default, and when you generate a new Rails app, it generates a layout for you.
```
Comment (AnnJohn) [inline README.md:114]: Can you include in this repo, a rails app that has layouts turned off, so when the clone this and play around in this rails app they will see what the problem is from the comment above around copy pasting stuff from index to show. Then in here, tell students to enable layouts and show how much better it is :)

File: README.md [hunk 2]
```diff
@@ -1,26 +1,197 @@
...
+
+To find the generated layout, go and have a look in your Rails app at the following path. When you render a template for an action without specifying a different layout to use, Rails will use the layout found at this location: **app/views/layouts/application.html.erb**.
+
+When you first generate a Rails app, this file will generate something very similar to this, depending on your version of Rails. The application.html.erb file is a very good place to start adding common components, like the navigation, search, and footer from the example above.
```
Comment (AnnJohn) [inline README.md]: backticks around `application.html.erb`

File: README.md [hunk 3]
```diff
@@ -1,26 +1,197 @@
...
+Without layouts you would need to copy and paste your logo, navigation, search, and footer code into each action's template that you want in your application.
+
+And when you want to make a change to one of the common components, for example the navigation, you would have to go and make that change in all the template files that you created in your application. This will obviously slow you down a lot.
+
+## Layouts To The Rescue
+
+Luckily you don't need to copy content from one template file to the next, because layouts in Rails are enabled by default, and when you generate a new Rails app, it generates a layout for you.
+
+To find the generated layout, go and have a look in your Rails app at the following path. When you render a template for an action without specifying a different layout to use, Rails will use the layout found at this location: **app/views/layouts/application.html.erb**.
+
```
Comment (AnnJohn) [inline README.md:117]: Nice. clear simple instructions

File: README.md [hunk 4]
```diff
@@ -1,26 +1,197 @@
...
+And this is the template for the about action, with a simple message in it, which we would want to display in our layout from above.
+
+```erb
+<!-- /app/views/static/about.html.erb -->
+
+<p>Hello!</p>
+```
+
+When you load up this route in your browser, you will be greeted by a very bold message saying: **Welcome To The Flatiron Store!**, but you won't see the **Hello!** from the about action's template.
```
Comment (AnnJohn) [inline README.md]: woot! showing the pain

File: README.md [hunk 5]
```diff
@@ -1,26 +1,197 @@
...
+</html>
+```
+
+Now when you hit up this route in your browser, you will see **Welcome To The ACME Store!**, followed by **Hello!**. This means that when the layout rendered, it pulled the action specific template into the correct place, where we added the `yield` statement.
+
+## How Layouts and Templates Are Stitched Together
+
+At it's simplest level, this is what happens when a request is made to your rails application:
+
+1. Rails finds the template for your action based on either convention, or any other options you passed to the render method in your controller action.
```
Comment (AnnJohn) [inline README.md:219]: For each step can you ground it in the concrete and make sure you just mention what file you are talking about from your previous flatiron example

File: README.md [hunk 6]
```diff
@@ -1,26 +1,197 @@
 # Rails Layouts And Templates
 
+This lesson will show you how to use layouts to achieve a common look and feel between different views in your app.
```
Comment (AnnJohn) [inline README.md:3]: Good succint overview

File: README.md [hunk 7]
```diff
@@ -1,26 +1,197 @@
...
+
 1. Identify the default application layout.
 2. Yield to view templates from a layout.
 3. Specify a custom layout in ActionController on a controller level using the `layout` macro and on the action level using the `render :layout => "custom"` option.
 4. Use :layout => false to shut off the layout and only render the view
 
-## Outline
+## Life Without Layouts
+
+Imagine that you're tasked to build an online store app with Rails.
```
Comment (AnnJohn) [inline README.md:16]: Love that you're immediately drawing the student in by using the second-person and by putting them in a scenario. Scenario-based learning is super effective.

File: README.md [hunk 8]
```diff
@@ -1,26 +1,197 @@
...
+
+You would probably have a few different views in this app, for example:
+
+1. A list of products
+2. A detail view that shows more info for a selected product
+3. A shopping cart
+
+Across all these views you would want a consistent look, which is called a layout. This layout perhaps contains something like a logo, navigation links, a search bar, and a footer at the bottom that contains some info about the shop.
+
```
Comment (AnnJohn) [inline README.md:25]: Good slow easing into layouts. You're using concrete, real-world concepts here to draw in

File: README.md [hunk 9]
```diff
@@ -1,26 +1,197 @@
...
+Without layouts you would need to copy and paste your logo, navigation, search, and footer code into each action's template that you want in your application.
+
+And when you want to make a change to one of the common components, for example the navigation, you would have to go and make that change in all the template files that you created in your application. This will obviously slow you down a lot.
+
```
Comment (AnnJohn) [inline README.md:111]: Nice elucidation of the problems that would occur if you didn't use layouts. This is great, but one strategy you can use if you'd like is to not mention the word layout at all. You can talk about the problems as you do here. And really show them the pain; i.e. Show the full HTML of a `show.html.erb` (with all the JS and `head` stuff) then show creating the HTML for the `index.html.erb` page. When you move from show to index, mention "ok so to get started we first copy paste all of the boiler plate code form show". Then mention "Wait a second, copy pasting isn't something we should be doing. That sounds like a task for computers to do. Layouts is the way we get around copy pasting the elements that exist in every page"

**Comment by blake41** [comment]
Looks good.  I fixed some small things and added a little commentary.  Going to merge this PR and then merge my changes in on top.

**Pull request closed without merging**